app/controllers/boxes_controller.rb

```diff
@@ -109,14 +110,12 @@ def load_box
   end
 
   def load_box_print
-    begin
-      @box = Box.where(institution: @navigation_context.institution).find(params.fetch(:id))
-    rescue ActiveRecord::RecordNotFound
-      @boxes = TransferPackage
-        .within( @navigation_context.institution )
-        .find( params.fetch( :transfer_package ) )
-        .boxes
-      @box = @boxes.find( params.fetch( :id ) )
+
+    @box = Box.where(institution: @navigation_context.institution, id: params.fetch(:id)).take
+    if @boxes.nil?
+      @box = Box
+        .joins(box_transfers: :transfer_package)
+        .where("boxes.id = ? and ( transfer_packages.sender_institution_id = ? or transfer_packages.receiver_institution_id = ? )", params.fetch(:id), @navigation_context.institution, @navigation_context.institution).take
```

Comment:
Because I was searching through the Transfer Packages model, and I removed the transfer_package id from all the logic. So I tried another way as I have just the box id and the institution.

Reply:
You can merge the `TransferPackage.within` scope with the Box relation:
```suggestion
        .merge(TransferPackage.within(@navigation_context.institution)).where(id: params.fetch(:id)), take
```